class.two-factor-core.php

```diff
@@ -495,15 +495,27 @@ public static function login_form_validate_2fa() {
 			$provider = self::get_primary_provider_for_user( $user->ID );
 		}
 
+		// Allow providers to re-send codes, etc.
+		if ( $provider->process_request( $user ) ) {
```

Comment:
We introduce `process_request()` for when we don't want to process the actual 2FA. However, we do this only after verifying the 2FA nonce.